src/engine.rs

```diff
@@ -1209,11 +1223,65 @@ impl Reedline {
                             }
                         }
                     }
-                    if self.editor.line_buffer().get_buffer().is_empty() {
+                    // Check if we should keep the menu active
+                    let buffer = self.editor.line_buffer().get_buffer();
+                    let should_stay_active = if self.auto_complete_menu {
+                        // Use same logic as auto-activate: need 1+ char after last space
+                        if let Some(last_word_start) = buffer.rfind(' ') {
+                            !buffer[last_word_start + 1..].is_empty()
+                        } else {
+                            // First word - keep menu if user activated it manually
+                            !buffer.is_empty()
+                        }
+                    } else {
+                        // No auto-complete mode, keep active unless empty
+                        !buffer.is_empty()
+                    };
+
+                    if !should_stay_active {
                         menu.menu_event(MenuEvent::Deactivate);
                     } else {
                         menu.menu_event(MenuEvent::Edit(self.quick_completions));
                     }
+                } else if self.auto_complete_menu {
+                    // Auto-activate completion menu (IDE style)
+                    // - Flags: show immediately on "-" or "--"
+                    // - Files/args: show after 1+ chars typed (not just space)
+                    // - Commands (first word): don't auto-complete
+                    let buffer = self.editor.line_buffer().get_buffer();
+                    let should_show = if let Some(last_word_start) = buffer.rfind(' ') {
+                        let last_word = &buffer[last_word_start + 1..];
+                        // Show if: flags (starts with -) OR has 1+ char typed
+                        !last_word.is_empty()
+                    } else {
+                        // First word - don't auto-complete commands
+                        false
+                    };
+
+                    if let Some(menu) = self.menus.iter_mut().find(|m| m.name() == "completion_menu") {
```

Comment:
Not a fan of the menu name being hardcoded here. I'd rather `auto_complete_menu` be an `Option<String>` holding the name of the menu to activate automatically. That way, users can choose which menu they want to always be active.